docs/docs/releases/0.4.0.md

```diff
@@ -0,0 +1,209 @@
+
+Special thanks to community member [@vishaljoshi789](https://github.com/vishaljoshi789) for the fix.
+
+*Related work:* [#4584](https://github.com/mathesar-foundation/mathesar/pull/4584 "Added table page column order on record pages")
+
+## Groundwork
+
+### Forms
+
+This release includes backend groundwork for a new Forms feature, which will allow users to collect structured data through public, shareable forms—without requiring logins or exposing the rest of a Mathesar instance.
+
+We’ve added core data models, API endpoints for creating, listing, replacing, and deleting forms, and early prototypes for how forms will appear on the Schema page. This functionality is hidden in 0.4.0 but will be published in an upcoming Mathesar release.
+
+*Related work:*
+
+[#4565](https://github.com/mathesar-foundation/mathesar/pull/4565 "Merge Forms feature branch into develop")
+[#4536](https://github.com/mathesar-foundation/mathesar/pull/4536 "Add `forms.add` RPC endpoint")
+[#4576](https://github.com/mathesar-foundation/mathesar/pull/4576 "Design prototype for listing Forms on the Schema page")
+[#4554](https://github.com/mathesar-foundation/mathesar/pull/4554 "Add form `get`, `list` & `delete` RPC endpoints")
+[#4577](https://github.com/mathesar-foundation/mathesar/pull/4577 "Add `forms.get_public` rpc endpoint")
+[#4525](https://github.com/mathesar-foundation/mathesar/pull/4525 "Form models")
+[#4642](https://github.com/mathesar-foundation/mathesar/pull/4642 "Hide forms for 0.4.0")
+[#4585](https://github.com/mathesar-foundation/mathesar/pull/4585 "Add `forms.replace` rpc endpoint")
+[#4568](https://github.com/mathesar-foundation/mathesar/pull/4568 "Ensure server integrity for forms")
+
+### Experimental "readonly" Mathesar role
+
+!!! warning "This feature is for advanced users"
+    This management command may cause unintended behavior and is not officially supported.
+
+We’ve added an experimental way to run Mathesar in full read-only mode. This allows for example, hosting a public-facing instance where users can browse and explore data without making any lasting changes, useful for sharing live demos or datasets safely. Setup involves creating a restricted database role and updating your deployment.
+
+??? info "Instructions for use (Please proceed with caution.)"
+    1. **Create a PostgreSQL database role** with read-only access to your user data.
+    2. **Create a Mathesar user** and assign it to that role.
+    3. **Log in as that user** to complete the first-time password setup.
+    4. **Run the management command** to restrict Django state:
+        ```bash
+        docker exec -it mathesar_service_dev python manage.py add_readonly_django -v 3 mathesar_ro_django
+        ```
+    5. **Update your `docker-compose.yml`** to make Mathesar use the read-only role for Django.
```

Comment:
Since you specified the role in the previous step, I'd do the same here: `mathesar_ro_django`.